### PR TITLE
[AArch64] Add codegen support to access thread local storage.

### PR DESCRIPTION
**Pull request by dave-estes**

### CODE REVIEW

File: hphp/runtime/vm/jit/code-gen-tls-arm.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+///////////////////////////////////////////////////////////////////////////////
+
+/*
+ * See detailed comment in code-gen-tls-x64.h.
+ */
+template<typename T>
+Vptr emitTLSAddr(Vout& v, TLSDatum<T> datum) {
+  uintptr_t vaddr = uintptr_t(datum.tls) - tlsBase();
+
+  auto b = v.makeReg();
```
Comment (mxw) [inline hphp/runtime/vm/jit/code-gen-tls-arm.h]: `auto const`

File: hphp/runtime/vm/jit/code-gen-tls-inl.h [hunk 2]
```diff
@@ -34,8 +34,9 @@ template<typename T>
 inline Vptr emitTLSAddr(Vout& v, TLSDatum<T> datum) {
   switch (arch()) {
     case Arch::X64:
-    case Arch::ARM:
       return x64::detail::emitTLSAddr(v, datum);
+    case Arch::ARM:
+       return arm::detail::emitTLSAddr(v, datum);
```
Comment (mxw) [inline hphp/runtime/vm/jit/code-gen-tls-inl.h]: nit: one space too many

Comment (dave-estes) [inline hphp/runtime/vm/jit/code-gen-tls-inl.h]: Done.

File: hphp/runtime/vm/jit/code-gen-tls-arm.h (hunk 1 above)
Comment (dave-estes) [inline hphp/runtime/vm/jit/code-gen-tls-arm.h]: Done.

**Comment by mxw** [comment]
Cool; landing.

**Pull request closed without merging**